extensions/binding/org.eclipse.smarthome.binding.fsinternetradio/src/main/java/org/eclipse/smarthome/binding/fsinternetradio/internal/FSInternetRadioHandlerFactory.java

```diff
@@ -35,13 +35,6 @@ public boolean supportsThingType(ThingTypeUID thingTypeUID) {
 
     @Override
     protected ThingHandler createHandler(Thing thing) {
-
-        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
-
-        if (thingTypeUID.equals(THING_TYPE_RADIO)) {
-            return new FSInternetRadioHandler(thing);
-        }
-
-        return null;
+        return new FSInternetRadioHandler(thing);
```

Comment:
@maggu2810 , you are right. But when I followed the code logic it turned out that  there is a method called `findThingHandlerFactory(ThingTypeUID thingTypeUID)` in the ThingManager class. And if the factory does not support the given ThingTypeUID, the createHandler method is never called. So in my opinion it can never return null, because the given ThingTypeUID is already checked.